spec/client/ics-008-wasm-client/README.md

```diff
@@ -141,42 +137,139 @@ interface Misbehaviour {
 
 ### Client initialization
 
-`Wasm Client` initialization requires a (subjectively chosen) latest consensus state and corresponding client state, interpretable by the `Wasm Contract`. 
+Wasm client initialization requires a (subjectively chosen) latest consensus state and corresponding client state, interpretable by the Wasm contract. 
 
 ```typescript
-  function initialize(cs: ClientState, clientStore: sdk.KVStore, state: exported.ConsensusState) {}
+function initialize(
+  identifier: Identifier,
+  data: []byte,
+  codeId: []byte,
+  consensusState: ConsensusState,
+  height: Height
+): ClientState {
+  provableStore.set("clients/{identifier}/consensusStates/{height}", consensusState)
+
+  // retrieve client identifier-prefixed store
+  clientStore = provableStore.prefixStore("clients/{identifier}")
+
+  // initialize wasm contract for a previously stored contract identified by codeId
+  initContract(codeId, clientStore)
+
+  return ClientState {
+    data,
+    codeId,
+    latestHeight: height
+  }
+}
 ```
 
 ### Validity predicate
 
-`Wasm Client` validity checking uses underlying `Wasm Contract`. If the provided header is valid, the client state will proceed to checking for misbehaviour and updating state.
+Wasm client validity checking uses underlying Wasm contract. If the provided client message is valid, the client state will proceed to checking for misbehaviour (call to `checkForMisbehaviour`) and updating state (call to either `updateStateOnMisbehaviour` or `updateState` depending on whether misbehaviour was detected in the client message). 
 
 ```typescript
-  function verifyClientMessage(cs: ClientState, clientStore: sdk.KVStore, clientMsg: exported.ClientMessage) {}
+function verifyClientMessage(clientMsg: ClientMessage) {
+  innerPayload = Map<string, any>()
```

Comment:
I added this pseudocode to kinda match what the implementation does (i.e. marshalling the payload as JSON). Let me know if you agree.